add info on scaling logic to typha overview (#1360)
tigera/docs: calico/reference/typha/overview.mdx

- Since one Typha instance can support hundreds of Felix instances, it reduces the load on the datastore by a large factor.
- Since Typha can filter out updates that are not relevant to Felix, it also reduces Felix's CPU usage. In a high-scale (100+ node) Kubernetes cluster, this is essential because the number of updates generated by the API server scales with the number of nodes.

> ## Operator Scaling Logic

Reviewer: ```suggestion
## Operator scaling logic
```
We use sentence case for headings.

> If you're using the Operator installation, then the operator will automatically scales Typha pods for optimal performance. 

Reviewer: ```suggestion
If you installed {{prodname}} using the operator, then the operator will automatically scale Typha pods to optimize performance. 
```

> - Small Cluster Considerations: Special handling is in place for clusters with fewer than five nodes.

Reviewer: * It's not clear what this special handling is.
* Whatever it is, we can keep the explanation as a regular sentence in the same paragraph as the previous sentence. For example, "For clusters with fewer than five nodes, we do things like this."
* If the special handling is merely what's described in the table, let's just let the table do the work. We can strike out this line.

> Here's the scaling logic it uses:

Reviewer: Remove. The table explains itself.

Typha Replicas per Node Count

| Nodes | Replicas | 
| ---- | ---- | 
| `1-2` | 1 |
| `3-4` | 2 |
| `<200` | 3 |
| `<400` | 4 |
| `<600` | 5 |
| `...` |   |
| `<2000` | 12 |
| `...` |   |
| `<3600` | 20 |

:::note

This algorithm may change as needs evolve.

> :::

Reviewer: On second thought, let's remove this too. The same applies to pretty much everything in the docs. If we make a change, that should be documented just like any other change.
